src/public-key/ed25519.lisp

```diff
@@ -36,20 +36,21 @@
 (defparameter *ed25519-digest* (make-digest :sha512))
 
 
-(defun ed25519-inv (x)
-  (declare (optimize (speed 3) (safety 0) (space 0) (debug 0))
-           (type integer x))
-  (modular-inverse (mod x +ed25519-q+) +ed25519-q+))
+(defmacro ed25519-inv (x)
```

Comment:
why not just (declaim inline) it before its definition instead of turning it into a macro?

that way the optimize declarations of the call site will apply to the inlined defun body (at least on SBCL).